packages/core/saved-objects/core-saved-objects-migration-server-internal/src/state.ts

```diff
@@ -183,6 +188,22 @@ export interface DoneState extends PostInitState {
   readonly controlState: 'DONE';
 }
 
+/**
+ * Compatibe migrations do not require migrating to a new index because all
+ * schema changes are compatible with current index mappings.
+ *
+ * Before running the compatible migration we need to prepare. For example, we
+ * need to make sure that no older Kibana versions are still writing to target
+ * index.
+ */
+export interface PrepareCompatibleMigration extends PostInitState {
+  /** We have found a schema-compatible migration, this means we can optimise our migration steps */
+  readonly controlState: 'PREPARE_COMPATIBLE_MIGRATION';
+  /** Alias-level actions that prepare for this migration */
+  readonly preTransformDocsActions: AliasAction[];
+  readonly sourceIndexMappings?: IndexMapping;
```

Comment:
It's not related to your PR but `sourceIndexMappings` seems completely unused (probably after https://github.com/elastic/kibana/pull/147296). I had to actually remove the code locally to make sure 😂 so I can submit that as a follow-up PR.

Reply:
I'll remove it